### PR TITLE
DACT-1543: building ch01 occupation json with validation

### PR DESCRIPTION
**Pull request by olaych**
Jira: https://companieshouse.atlassian.net/browse/DACT-1543
- Added occupation update validation
- Added occupation ch01 json build
- Added test

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerUpdateValidator.java [hunk 1]
```diff
@@ -154,6 +155,20 @@ public void validateNationalitySection(HttpServletRequest request, List<ApiError
         validateNationalityLength(request, errorList, dto);
     }
 
+    public void validateOccupationSection(HttpServletRequest request, List<ApiError> errorList, OfficerFilingDto dto, AppointmentFullRecordAPI appointmentFullRecordAPI) {
+        if (Boolean.FALSE.equals(dto.getOccupationHasBeenUpdated())) {
+                return;
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerUpdateValidator.java:160]: ident

File: src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerUpdateValidator.java [hunk 2]
```diff
@@ -154,6 +155,20 @@ public void validateNationalitySection(HttpServletRequest request, List<ApiError
...
+        }
+        if (dto.getOccupation() == null) {
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerUpdateValidator.java:162]: Does a null occupation not mean we are removing the occupation? So we still need to check against CHIPS?

Comment (AndreKramer7654) [reply src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerUpdateValidator.java:162]: or have we set it to NONE already?

**Review by jbestwickch — DISMISSED** [review]

File: src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerUpdateValidator.java (hunk 2 above)
Comment (olaych) [reply src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerUpdateValidator.java:162]: Null and None against chips data is checked doesOccupationMatchChipsData()

File: src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerUpdateValidator.java [hunk 3]
```diff
@@ -175,6 +190,15 @@ public boolean doesNationalityMatchChipsData(OfficerFilingDto dto, AppointmentFu
         return chipsNationalities.length < 3 || matchesChipsField(dto.getNationality3(), chipsNationalities[2]);
     }
 
+    public boolean doesOccupationMatchChipsData(OfficerFilingDto dto, AppointmentFullRecordAPI appointmentFullRecordAPI) {
+        if (appointmentFullRecordAPI.getOccupation() == null ||
+                appointmentFullRecordAPI.getOccupation().equalsIgnoreCase("NONE") && !dto.getOccupation().equalsIgnoreCase("NONE")) {
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerUpdateValidator.java:195]: dto.getOccuption() could be "" or null and that means same as "NONE" so:
... && (!dto.getOccupation().equalsIgnoreCase("NONE") || dto.isEmpty())

**Review by clewis1 — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerUpdateValidator.java (hunk 2 above)
Comment (AndreKramer7654) [reply src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerUpdateValidator.java:162]: But it's the same a NONE so should not be a change? We need to still cann doesOccupationMatchChips data and make that treat "",null and "NONE" all as the same.

**Review by AndreKramer7654 — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/officerfiling/api/service/FilingDataServiceImpl.java [hunk 4]
```diff
@@ -223,15 +226,19 @@ public Boolean mapCorporateDirector(Transaction transaction, AppointmentFullReco
...
         var officerFilingName = "";
         if (companyAppointment.getSurname() != null) {
-            officerFilingName = getNameFromAppointment(companyAppointment);
+            surname = companyAppointment.getSurname().toUpperCase();
+            middleNames = companyAppointment.getOtherForenames();
+            officerFilingName = companyAppointment.getForename() + " " + middleNames + " " + surname;
         } else {
             // is a corporate director
-            officerFilingName = companyAppointment.getName().toUpperCase();
+            officerFilingName = companyAppointment.getName();
```
Comment (jbestwickch) [inline src/main/java/uk/gov/companieshouse/officerfiling/api/service/FilingDataServiceImpl.java:241]: Not clear why changes in this method are being made?

Comment (olaych) [reply src/main/java/uk/gov/companieshouse/officerfiling/api/service/FilingDataServiceImpl.java]: Yea had merge issues yesterday but now reverted changes to main

**Review by mdesaikns — APPROVED** [review]

**Review by PhilipGlover — APPROVED** [review]
lgtm

**Pull request merged**